Quick start for existing users v2 Delete - "edit your homepage" tour

AjeshRPai opened the pull request:
Fixes #16782 
Dependant fluxC PR - https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2446

Internal reference: Slack thread - p1650367281046699

This PR removes the quick start task tour Edit your home page from the app. 

---
## To test:

### Test 1
#### No pending tasks are completed after migration 

1. Install a build having Edit your home page - For eg build from this [Latest merged PR to trunk](https://github.com/wordpress-mobile/WordPress-Android/pull/16786)
2. log in to the app 
3. Create a site and start quick start 
4. Notice the pending tasks 
5. Install the build from this PR 
6. Notice that the pending task is not completed 
7. Notice that Edit your home page task is not visible 

| Before  | After Removing Edit your home page |
|---|---|
|![Screenshot_20220622-162250](https://user-images.githubusercontent.com/17463767/175015310-c8bceba8-8d9c-485f-9391-b29178c3c8ad.jpg)|![Screenshot_20220622-162803](https://user-images.githubusercontent.com/17463767/175015515-65a99933-aeb1-4019-8b16-db51991d800e.jpg)| 

### Test 2 
#### When customise your site has any pending task other than Edit your home page 

1. Install a build having Edit your home page - For eg build from this [Latest merged PR to trunk](https://github.com/wordpress-mobile/WordPress-Android/pull/16786)
2. log in to the app 
3. Create a site and start quick start 
4. Finish Edit your home page task and make sure you have any other pending task 
5. Install the build from this PR 
6. Notice that the pending task is not completed 
7. Notice that Edit your home page task is not visible 

| Before  | After Removing Edit your home page |
|---|---|
|![Screenshot_20220622-122908](https://user-images.githubusercontent.com/17463767/174998813-63090f7e-b17a-4a98-88c6-bed70424d2ed.jpg)|![Screenshot_20220622-123021](https://user-images.githubusercontent.com/17463767/174998826-e39fba71-d4a7-4522-be5f-fcbf2573a4e8.jpg)| 

### Test 3

#### When only Edit your home page task is remaining in customise your site and Grow your audience tasks are pending 

1. Install a build having Edit your home page 
2. log in to the app 
3. Create a site and start quick start 
4. Finish all tasks in customise your site except **Edit your home page** 
5. Install the build from this PR 
6. Notice that all tasks in customise your site is completed 
7. Notice that customise your site is struck in quick start card 
9. Notice that Edit your home page task is not visible 
10. Notice that No tasks from Grow your audience is completed 

| Before Quick start Task list | Before |  After  | After Quick start task list |
|---|---|---|---|
|![Screenshot_20220622-123359](https://user-images.githubusercontent.com/17463767/174998831-2ae7f94f-812a-4d04-94bd-23d344616514.jpg)|![Screenshot_20220622-123406](https://user-images.githubusercontent.com/17463767/174998835-91d0b8dd-c86d-46b2-9db1-c9ef91abc956.jpg) |![Screenshot_20220622-123436](https://user-images.githubusercontent.com/17463767/174998837-4e1c082a-c6ce-428b-9109-02a10339633e.jpg)|![Screenshot_20220622-123439](https://user-images.githubusercontent.com/17463767/174998839-81a04826-bad9-484f-877d-85c77acf68d1.jpg)|

### Test 4 
#### When only Edit your home page task is remaining in quick start list 

1. Install a build having Edit your home page 
2. log in to the app 
3. Create a site and start quick start 
4. Finish all quick start tasks  except **Edit your home page** 
5. Install the build from this PR 
6. Notice that quick start card is not visible 
7. Verify that Notification with quick start completed message is shown in system notifications and notifications tab  
9. Notice that Edit your home page task is not visible 

| Before  | Before Task list | After - Notifications  | After - Notifications tab |
|---|---|---|---|
|![Screenshot_20220622-125323](https://user-images.githubusercontent.com/17463767/174998841-aaf6103e-c51a-43a5-81e2-352f2784b1b8.jpg)|![Screenshot_20220622-125327](https://user-images.githubusercontent.com/17463767/174998842-992a46fd-b3c3-40c1-be32-27dabcf97b85.jpg)|![Screenshot_20220622-125731](https://user-images.githubusercontent.com/17463767/174998844-1de26104-8f0d-4a20-94f9-7de042e1f80f.jpg)|![Screenshot_20220622-125740](https://user-images.githubusercontent.com/17463767/174998850-93d81c05-1110-41f6-8af2-82e709bf5594.jpg)|

Note: One of the two notifications for the quick start completed is from the playstore build, so please ignore

### Test 5
#### Verify that only code related to the Edit your home page task are deleted 
1. Verify focus point logic is deleted 
2. Verify Analytics tracking related to edit your home page is deleted
3. Verify ui logic related to edit your home page is deleted  


### Merge Instructions 
1. Make sure dependant flux c PR - https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2446 is merged 
2. Update fluxc version in build.gradle 
3. Remove Not ready for merge label 
4. Merge as normal

## Regression Notes
1. Potential unintended areas of impact
Quick start task list gets updated wrong after migration

5. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

6. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

AjeshRPai committed:
Updates: build gradle to latest fluxc version

AjeshRPai committed:
Removes: edit your home page task from quick start repository

AjeshRPai committed:
Removes: Edit your home page Quick start notice details and resources

AjeshRPai committed:
Removes: Edit your home page Quick start task details

AjeshRPai committed:
Removes: Redundant resources

AjeshRPai committed:
Removes: edit home page focus point logic

AjeshRPai committed:
Removes: quick start edit home page logic from card source

AjeshRPai committed:
Removes: quick start edit home page details from builder,proompt and vm

AjeshRPai committed:
Removes: quick start edit home page logic from pages fragment

AjeshRPai committed:
Removes: redundant quick start edit home page tracking

AjeshRPai committed:
Fixes: Lint issues

AjeshRPai committed:
Removes: redundant test and dependency from test class

AjeshRPai committed:
Fixes: Lint issues

AjeshRPai committed:
Removes: redundant tests and fixes failing tests

AjeshRPai committed:
Removes: redundant resource

AjeshRPai committed:
Reverts: accidental changes to Project.xml

ashiagr commented:
Thank you for this PR, @AjeshRPai! I appreciate that you included several tests but `Quick Start` is a tricky feature, 😅 I was able to find one issue:

- Complete all tasks (including `Edit your homepage`) in both `grow` and `customize` types except one task e.g. `View your site` in a build where `Edit your homepage` task is present.
- Install build from this PR.
- Notice that completion notification is shown even when one task `View your site` was not complete.

This is because we've specified an incorrect task type `grow` for `edit_homepage` task in `FluxC` SQL migration. 

Everything else looks great! Let me know once that is fixed and I'll approve the PR.

AjeshRPai committed:
Updates: FluxC Version

AjeshRPai commented:
@ashiagr Thanks for testing the changes thoroughly and finding the issue. 🙌🏼 👍🏼 . I have fixed the issue in the dependant fluxc PR with this [commit](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2446/commits/08115e553dcf3440a807d7df1d0d84e9e54e1c83) and updated this PR(commit [99d6119](https://github.com/wordpress-mobile/WordPress-Android/pull/16790/commits/99d6119fc5b149886c2e0d017265fda6cf30af62)) with fluxC Version pointing to that changes. Please have a look at this PR again.

ashiagr committed:
Update FluxC version

ashiagr reviewed (approved):
LGTM 👍
